### PR TITLE
feat(middleware): process feed title and description using LLM

### PR DESCRIPTION
**Pull request by coxde**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
NOROUTE
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
Add a title translation feature using LLM, I'll open a new PR later for the [docs](https://github.com/RSSNext/rsshub-docs).

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
This can be done by modifying https://github.com/DIYgod/RSSHub/blob/c2937395f1e1e2bd70a6b649cfcc94cb4d4096ff/lib/middleware/parameter.ts#L348-L372
to pass the title as well instead of adding a new common parameter

**Comment by coxde** [comment]
> This can be done by modifying
> 
> https://github.com/DIYgod/RSSHub/blob/c2937395f1e1e2bd70a6b649cfcc94cb4d4096ff/lib/middleware/parameter.ts#L348-L372
> 
> to pass the title as well instead of adding a new common parameter

Indeed. Maybe I could also provide an option for passing all, title only or description only? For my case, I just need to process the title.

**Comment by TonyRL** [comment]
You may add an extra env like `config.openai.inputOption` which takes either `title`, `description` or `both` with `description` being the default one so it won't break the current flow.

**Comment by coxde** [comment]
> You may add an extra env like `config.openai.inputOption` which takes either `title`, `description` or `both` with `description` being the default one so it won't break the current flow.

I've done the refactoring, feel free to check out any further improvements as my knowledge of JavaScript is limited.

Besides `OPENAI_INPUT_OPTION`, I also separated the prompts for title and description, which might be useful for `both` mode. I kept the original prompt for compatibility. But this is an example that works best in my case.
```
OPENAI_PROMPT_TITLE='Please translate the following title into Simplified Chinese. Output ONLY translated text.'
OPENAI_PROMPT=Please summarize the following article CONCISELY into Simplified Chinese and reply with markdown format. Output ONLY summarized text.
```

File: lib/middleware/parameter.ts [hunk 1]
```diff
@@ -327,23 +327,53 @@ const middleware: MiddlewareHandler = async (ctx, next) => {
...
-                                }
-                                const summary_md = await summarizeArticle(text);
-                                return md.render(summary_md);
+                    try {
+                        // handle description
+                        if (config.openai.inputOption === 'description' && item.description) {
+                            const description = await cache.tryGet(`openai:description:${item.link}`, async () => {
+                                const description = convert(item.description!);
+                                const description_md = await getAiCompletion(config.openai.promptDescription, description);
+                                return md.render(description_md);
```
Comment (TonyRL) [inline lib/middleware/parameter.ts:336]: `camelCase`

File: lib/middleware/parameter.ts [hunk 2]
```diff
@@ -327,23 +327,53 @@ const middleware: MiddlewareHandler = async (ctx, next) => {
...
+                            });
+                            // replace the title
+                            if (title !== '') {
+                                item.title = title + '';
+                            }
+
+                            const description = await cache.tryGet(`openai:description:${item.link}`, async () => {
+                                const description = convert(item.description!);
+                                const description_md = await getAiCompletion(config.openai.promptDescription, description);
+                                return md.render(description_md);
```
Comment (TonyRL) [inline lib/middleware/parameter.ts:368]: `camelCase`

**Review by TonyRL — COMMENTED** [review]
Please update the test case as well

**Comment by coxde** [comment]
> Please update the test case as well

Could you kindly take a look? Not sure why there are two coverage missing.

**Pull request merged**